fix: registration query schema (#2)
namehash/ens-referrals: README.md

> # ENS Referral Program

Reviewer: ```suggestion
# ENS Referral Program: Developer Resources
```

> [ENS Referral Program](https://ensawards.org/ens-referral-program) is an incentive framework designed to distribute awards for referred .eth name registrations and renewals. It is currently being rolled out as a series of time-limited editions, with the long-term goal of refining its rules and mechanisms before launching a continuous, indefinite program.

Reviewer: ```suggestion
The [ENS Referral Program](https://ensawards.org/ens-referral-program) is an incentive framework designed to distribute awards for referred .eth name registrations and renewals. It is being rolled out as a series of time-limited editions to refine its rules and mechanisms, with the goal of launching a long-term ENS Referral Program.
```

> So far, we have launched the `ENS Holiday Awards` edition (see details [here](https://ensawards.org/ens-referral-program/editions/2025-12/rules)) and are preparing more refined editions for `April 2026` and `May 2026`. Additional details will be announced soon.

Reviewer: ```suggestion
So far, we have launched the `ENS Holiday Awards` edition (see details [here](https://ensawards.org/ens-referral-program/editions/2025-12/leaderboard)) and are preparing more refined editions for `April 2026` and `May 2026`. Additional details will be announced soon.
```

These efforts are part of our SPP2 deliverables to the ENS DAO, which include our commitment to ultimately fund $50,000 USD in ENS Referral Program awards throughout the duration of SPP2.

## Developer Integration Guide

Developers can participate in any ENS Referral Program edition through referral hyperlinks or through onchain referrals at the smart-contract level.

> ### Referral Hyperlinks

Reviewer: Suggest to update the link below to be https://ensawards.org/ens-referral-program#generate-referral-link

Integrate directly with ENS smart contracts for full flexibility and control over the user journey. Build the ultimate .eth name registration and renewal features directly into your app or wallet.

> Onchain referrals for the registration / renewal of a direct subname of ".eth" should be implemented using the following contracts within the duration of any ENS Referral Program edition:

Reviewer: ```suggestion
Onchain referrals for the registration / renewal of a direct subname of ".eth" should be implemented using the following contracts within the duration of any of the announced ENS Referral Program editions:
```

* [Sepolia deployment](https://sepolia.etherscan.io/address/0x7AB2947592C280542e680Ba8f08A589009da8644) (useful for testing, but no ENS Referral Program award distributions will be made on Sepolia)

#### "referrer" parameter values

The "referrer" parameter passed to the register / renew calls on the above contracts is a 32-byte value, while an Ethereum address is a 20-byte value. Therefore, to qualify for this program, the "referrer" parameter must match the following format:
* First 12-bytes: all zeros.
* Last 20-bytes: Ethereum mainnet address identifying the Referrer and where any Award Distributions the Referrer may earn will be deposited.
* Related utility functions for encoding / decoding referrer address values for use onchain can be found [here](https://github.com/namehash/ensnode/blob/main/packages/ensnode-sdk/src/registrars/encoded-referrer.ts).

### ENS Referral Program Rule Implementation

Logic implementing the `ENS Referral Program Rules` (all editions) is fully open source and can be found [here](https://github.com/namehash/ensnode/tree/main/packages/ens-referrals/src/v1).

### ENS Referral Program APIs

> [ENSNode](https://ensnode.io/) v1.7.0+ offers APIs for querying ENS Referral Program leaderboards, referral metrics for specific referrers, and overall editions information. The easiest way to query these APIs is through the ENSNode client library as seen [here](https://github.com/namehash/ensnode/tree/main/packages/ens-referrals/src/v1). For more information [read the client docs](https://github.com/namehash/ensnode/blob/main/packages/ens-referrals/README.md).

Reviewer: ```suggestion
[ENSNode](https://ensnode.io/) v1.7.0+ offers APIs for querying ENS Referral Program leaderboards, referral metrics for specific referrers, and overall editions information. The recommended way to query these APIs is through the [ENS referrals client library] (https://github.com/namehash/ensnode/blob/main/packages/ens-referrals/README.md).
```
